lldb/source/Plugins/Process/Dpu/DpuRank.cpp

```diff
@@ -115,22 +117,60 @@ void DpuRank::SetSliceInfo(uint32_t slice_id, uint64_t structure_value,
                                            slice_target);
 }
 
-Dpu::Dpu(DpuRank *rank, dpu_t *dpu) : m_rank(rank), m_dpu(dpu) {
+Dpu::Dpu(DpuRank *rank, dpu_t *dpu, FILE *stdout_file_)
+    : m_rank(rank), m_dpu(dpu), printf_enable(false),
+      printf_buffer_last_addr((uint32_t)LLDB_INVALID_ADDRESS),
+      printf_buffer_var_addr((uint32_t)LLDB_INVALID_ADDRESS) {
   nr_threads = m_rank->GetNrThreads();
   nr_of_work_registers_per_thread =
       rank->GetDesc()->dpu.nr_of_work_registers_per_thread;
 
   m_context = dpu_alloc_dpu_context(dpu_get_rank(dpu));
-}
 
-Dpu::~Dpu() { dpu_free_dpu_context(m_context); }
+  stdout_file = stdout_file_;
+}
+
+Dpu::~Dpu() {
+  fclose(stdout_file);
+  dpu_free_dpu_context(m_context);
+}
+
+bool Dpu::GetPrintfSequenceAddrs() {
+  dpu_runtime_context_t *runtime = dpu_get_runtime_context(m_dpu);
+  open_print_sequence_addr = runtime->open_print_sequence_addr;
+  close_print_sequence_addr = runtime->close_print_sequence_addr;
+  printf_buffer_address = runtime->printf_buffer_address;
+  printf_buffer_size = runtime->printf_buffer_size;
+  printf_buffer_var_addr = runtime->printf_write_pointer_address;
+  if (open_print_sequence_addr != (lldb::addr_t)LLDB_INVALID_ADDRESS &&
+      close_print_sequence_addr != (lldb::addr_t)LLDB_INVALID_ADDRESS) {
+    open_print_sequence_addr++; // we add 1 to avoid to be on the 'acquire'
+                                // which could makes us loop forever
+    open_print_sequence_addr *= 8;
+    close_print_sequence_addr *= 8;
```

Comment:
Maybe add a function to translate pc to IRAM byte address ?

Reply:
Done